app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class AnnouncementsController < ApplicationController
+  def index
+    @announcements = Announcement.all
+    @new_announcements = Announcement.new
+  end
+
+  def show
+    @announcement = Announcement.find params[:id]
+  end
+
+  def create
+    announcement = Announcement.new announcement_params
+    announcement.date = Time.zone.now
```

Comment:
You don't need a `date` column actually. Rails has a `created_at` by default, can use that